org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/TryStatement.java

```diff
@@ -583,7 +587,14 @@ public void generateCode(BlockScope currentScope, CodeStream codeStream) {
 				}
 			}
 		}
-		this.tryBlock.generateCode(this.scope, codeStream);
+		if (this.addPatternAccessorException)
+			codeStream.addPatternCatchExceptionInfo(this.tryBlock.scope, this.recPatCatchVar);
+
+		this.tryBlock.generateCode(this.tryBlock.scope, codeStream);
```

Comment:
This looks wrong to me. I see no reason to change the scope. I suspect this is causing some of the failures in the latest test run.

Reply:
Just to be clear, the point is about the scope argument. Before the patch, it was `this.scope` but now it is `this.tryBlock.scope`.